### PR TITLE
Fixes #19093: Parse id of method calls with rudderc

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/19093

### CODE REVIEW

File: rudder-lang/tests/techniques/supported_formats/technique.rl.cf [hunk 1]
```diff
@@ -15,43 +15,43 @@ bundle agent supported_formats_technique {
...
                                          unless => concat("windows");
-    "Directory present_${report_data.directive_id}_2" usebundle => log_rudder("Skipping method 'Directory present' with key parameter 'shared_cf_dsc_condition' since condition 'windows' is not reached", "shared_cf_dsc_condition", canonify("${class_prefix}_directory_present_shared_cf_dsc_condition"), canonify("${class_prefix}_directory_present_shared_cf_dsc_condition"), @{args}),
+    "directory_present_${report_data.directive_id}_2" usebundle => log_rudder("Skipping method 'directory_present' with key parameter 'shared_cf_dsc_condition' since condition 'windows' is not reached", "shared_cf_dsc_condition", canonify("${class_prefix}_directory_present_shared_cf_dsc_condition"), canonify("${class_prefix}_directory_present_shared_cf_dsc_condition"), @{args}),
                                          unless => concat("windows");
 
     # Registry key present:
     # 
-    #   registry("DSC_ONLY").key_present() as registry_key_present_DSC_ONLY
+    #   registry_key("""DSC_ONLY""").present() as registry_key_present_DSC_ONLY
```
Comment (amousset) [inline rudder-lang/tests/techniques/supported_formats/technique.rl.cf:45]: why the triple quotes?

File: rudder-lang/src/technique.rs [hunk 2]
```diff
@@ -220,8 +221,12 @@ pub struct MethodCall {
     condition: String,
     method_name: String,
     component: Option<String>,
+    #[serde(default = "generate_id")] // >=7.0
     id : String,
 }
+fn generate_id () -> String {
```
Comment (amousset) [inline rudder-lang/src/technique.rs:227]: could you run `cargo fmt` on your project?

File: rudder-lang/tests/techniques/supported_formats/technique.rl.cf (hunk 1 above)
Comment (VinceMacBuche) [reply rudder-lang/tests/techniques/supported_formats/technique.rl.cf:45]: i did not touched that file ... i don't know how where they modified

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

File: rudder-lang/src/technique.rs (hunk 2 above)
Comment (VinceMacBuche) [reply rudder-lang/src/technique.rs]: done ! i did not know this pro-tip

**Review by amousset — APPROVED** [review]

**Pull request merged**